src/main/java/com/opentable/kafka/embedded/EmbeddedKafkaBroker.java

```diff
@@ -141,12 +146,29 @@ public void createTopic(String topic) {
     }
 
     public <K, V> KafkaProducer<K, V> createProducer(Class<? extends Serializer<K>> keySer, Class<? extends Serializer<V>> valueSer) {
+        Properties props = baseProducerProperties();
+        props.put("key.serializer", keySer.getName());
+        props.put("value.serializer", valueSer.getName());
+        return new KafkaProducer<>(props);
+    }
```

Comment:
I do like that you're pushing for more static safety by having the programmer pass in the `Class` instead of just a string name.  Let's also support the constructor that takes serializer and deserializer instances.